File: LeetCode_3577/Main.py
class Solution(object):
    def countPermutations(self, complexity):
        """
        :type complexity: List[int]
        :rtype: int
        """
        MOD = 10**9 + 7
        n = len(complexity)
        
        # Check unlock possibility
        root = complexity[0]
        for i in range(1, n):
            if complexity[i] <= root:
                return 0   # cannot unlock computer i
        
        # Compute (n-1)!
        res = 1
        for i in range(1, n):
            res = (res * i) % MOD
        
        return res

File: LeetCode_3577/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int countPermutations(vector<int>& c) {
        const int MOD = 1e9+7;
        int n = c.size();
        
        // If any complexity[i] <= complexity[0], impossible to unlock
        for (int i = 1; i < n; i++) {
            if (c[i] <= c[0]) return 0;
        }

        // Compute (n-1)!
        long long res = 1;
        for (int i = 1; i < n; i++) {
            res = (res * i) % MOD;
        }
        return res;
    }
};
